host/src/raiko.rs

```diff
@@ -201,10 +193,7 @@ impl Prover for NativeProver {
 
 fn check_eq<T: std::cmp::PartialEq + std::fmt::Debug>(expected: T, actual: T, message: &str) {
     if expected != actual {
-        println!(
-            "Assertion failed: {} - Expected: {:?}, Found: {:?}",
-            message, expected, actual
-        );
+        debug!("Assertion failed: {message} - Expected: {expected:?}, Found: {actual:?}");
```

Comment:
err or warn??

Reply:
I'll make it warn since it doesn't seem to block execution (if that is expected behavior). If it should block, I'd use an error here.